crates/languages/src/c.rs

```diff
@@ -292,6 +292,40 @@ impl super::LspAdapter for CLspAdapter {
         }
         Ok(original)
     }
+
+    fn process_diagnostics(
+        &self,
+        params: &mut lsp::PublishDiagnosticsParams,
+        buffer_access: Option<(LanguageServerId, &'_ Buffer)>,
```

Comment:
```suggestion
        server_id: LanguageServerId,
        buffer: Option<&'_ Buffer>,
```

`LanguageServerId` is not about buffers, it's about that `lsp::PublishDiagnosticsParams` above and should always be present, as the message emitter's id.